firmware/version_data.py

```
@@ -0,0 +1,99 @@
+#!/usr/bin/env python
+
+import subprocess
+import os
+import tempfile
+import filecmp
+import shutil
+
+commit = subprocess.check_output(['git', 'log', '--format="%H"', '-n', '1'])
+print("Showing commit variable")
+print(commit[1:-2])
+
+branch = subprocess.check_output(['git', 'symbolic-ref', '--short', 'HEAD'])
+print("Showing branch variable")
+print(branch[:-1])
+
+describe = subprocess.check_output(['git', 'describe', '--dirty'])
+print("Showing describe variable")
+print(describe[:-1])
+
+status = subprocess.check_output(['git', 'status', '--short'])
+length = status.count(b'\n')
+print(status)
+
+print("Showing uplatform")
+if "PLATFORM" in os.environ:
+    platform = os.environ['PLATFORM']
+    print(platform.upper())
+else:
+    platform = ""
```

I think you should probably raise an error here with a useful message.